nodejs/aws-infra/experimental/clusterAutoScaling.ts

```diff
@@ -0,0 +1,566 @@
+
+        const parentOpts = { parent: this };
+
+        const cluster = args.cluster;
+
+        // Create the full name of our CloudFormation stack here explicitly. Since the CFN stack
+        // references the launch configuration and vice-versa, we use this to break the cycle.
+        // TODO[pulumi/pulumi#381]: Creating an S3 bucket is an inelegant way to get a durable,
+        // unique name.
+        const stackName = pulumi.output(args.stackName!) || new aws.s3.Bucket(name, {}, parentOpts).id;
+
+        // Use the instance provided, or create a new one.
+        const instanceProfile = getInstanceProfile(name, args, parentOpts);
+        const fileSystem = args.fileSystem;
+
+        const instance = new aws.ec2.LaunchConfiguration(
+            name, {
+            ...args,
+            imageId: getEcsAmiId(args.ecsOptimizedAMIName),
+            instanceType: pulumi.output(args.instanceType).apply(t => t || "t2.micro"),
+            keyName: args.keyName,
+            iamInstanceProfile: instanceProfile.id,
+            enableMonitoring: pulumi.output(args.enableMonitoring).apply(b => b !== undefined ? b : true),
+            placementTenancy: pulumi.output(args.placementTenancy).apply(t => t || "default"),
+            rootBlockDevice: pulumi.output(args.rootBlockDevice).apply(d => d || defaultRootBlockDevice),
+            ebsBlockDevices: pulumi.output(args.ebsBlockDevices).apply(d => d || defaultEbsBlockDevices),
+            securityGroups: pulumi.output(args.securityGroups).apply(g => g || [ cluster.instanceSecurityGroup.id ]),
+            userData: getInstanceUserData(cluster, args, stackName),
+        }, parentOpts);
+
+        this.instance = instance;
+        this.cluster = cluster;
+        this.stackName = stackName;
+        this.instanceProfile = instanceProfile;
+        this.fileSystem = fileSystem;
+
+        this.registerOutputs({
+            instance,
+            cluster,
+            stackName,
+            instanceProfile,
+            fileSystem,
+        });
+    }
+}
+
+(<any>ClusterAutoScalingLaunchConfiguration).doNotCapture = true;
+
+const defaultAssumeInstanceRolePolicyDoc: aws.iam.PolicyDocument = {
+    Version: "2012-10-17",
+    Statement: [{
+        Action: [
+            "sts:AssumeRole",
+        ],
+        Effect: "Allow",
+        Principal: {
+            Service: [ "ec2.amazonaws.com" ],
+        },
+    }],
+};
+
+const defaultRootBlockDevice = {
+    volumeSize: 8, // GiB
+    volumeType: "gp2", // default is "standard"
+    deleteOnTermination: true,
+};
+
+const defaultEbsBlockDevices = [{
+        // Swap volume
+        deviceName: "/dev/xvdb",
+        volumeSize: 5, // GiB
+        volumeType: "gp2", // default is "standard"
+        deleteOnTermination: true,
+    }, {
+        // Docker image and metadata volume
+        deviceName: "/dev/xvdcz",
+        volumeSize: 50, // GiB
+        volumeType: "gp2",
+        deleteOnTermination: true,
+    }];
```

Comment:
this should be exposed in some way.  that way if the user wants these, but wants to add another device, they can do so without having to copy this all over into their code.